packages/headless/src/features/facets/facet-set/facet-set-selectors.ts

```diff
@@ -7,18 +7,21 @@ import {FacetResponse, FacetValue} from './interfaces/response';
 import {AnyFacetResponse} from '../generic/interfaces/generic-facet-response';
 
 export const baseFacetResponseSelector = (
-  state: SearchSection | ProductListingSection,
+  state: Partial<SearchSection | ProductListingSection>,
   id: string
 ) => {
-  if ('productListing' in state) {
+  if ('productListing' in state && state.productListing) {
```

Comment:
I think the following would be easier to read:
```suggestion
  if (state?.productListing) {
```

Reply:
It is, but the `in` operator is used by typescript to narrow its type to being a `ProductListingSection` type inside the if. 

I was actually contemplating to use a type predicate function for readability, like `isProductListingSection`.